api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -248,6 +252,15 @@ public DbUser updateProfile(
     user.setAreaOfResearch(updatedProfile.getAreaOfResearch());
     user.setProfessionalUrl(updatedProfile.getProfessionalUrl());
     user.setAddress(addressMapper.addressToDbAddress(updatedProfile.getAddress()));
+
+    boolean enableInitialCreditsExpiration =
+        configProvider.get().featureFlags.enableInitialCreditsExpiration;
+
+    if (enableInitialCreditsExpiration) {
+      initialCreditsExpirationService.setInitialCreditsExpirationBypassed(
```

Comment:
thinking out loud here... this will start the initial credits expiration clock for the user in some cases.  is that ok?

if bypassed = false -> we intend the user to be subject to the expiration clock.  So I think that's fine.

bypassed = true -> the user isn't subject to the clock, so it doesn't matter.

I guess the edge case is:
1. bypassed
2. later unbypassed

but that seems like it would not cover a large number of users.

Reply:
I am curious about that case too. I'll reach out to Moira and see if she has thoughts on it. Thanks for bringing it up.